templates/part.content.php

```diff
@@ -88,6 +88,78 @@ class="icon-toggle keep-unread"
+                                        <input
+                                            ng-model="nameQuery"
+                                            ng-model-options="{debounce: 400}"
+                                            ng-change="Share.searchUsers(nameQuery)"
+                                            type="text"
+                                            placeholder="<?php p($l->t('Username')) ?>"
+                                            title="<?php p($l->t('Username')) ?>"
+                                            name="contactName"
+                                            required
+                                            style="width: 200px">
+                                            <div ng-if="App.loading.isLoading('user')"
+                                                ng-class="{'icon-loading-small': App.loading.isLoading('user') }">
+                                            </div>
+                                    </fieldset>
+                                </form>
+
+                                <div style="margin-left: 2em; font-size: 0.85em;"
+                                     ng-if="!(Share.userList.length > 0) && nameQuery && !App.loading.isLoading('user')">
+                                    <?php p($l->t('No users found')) ?>
+                                </div>
+                                <a
+                                    class="icon-category-installed pr-3"
+                                    ng-repeat="user in Share.userList"
+                                    ng-click="Share.shareItem(item.id, user.value.shareWith)">
+                                    {{ user.value.shareWith }}
+                                    <span class="right" style="margin-top: 1.4em; margin-right: 1em"
+                                        ng-class="{'icon-loading-small': App.loading.isLoading(user.value.shareWith), 'icon-checkmark': !App.loading.isLoading(user.value.shareWith) && Share.usersSharedArticles[item.id].includes(user.value.shareWith)}">
+                                    </span>
+                                </a>
+                                <p class="label-group"> <?php p($l->t('Share on social media')) ?> </p>
+                            <div class="row">
+                                <div class="col-4">
+                                    <a target="_blank"
+                                        class="icon-dropdown icon-facebook pr-5"
+                                        ng-href="https://www.facebook.com/sharer/sharer.php?u={{ ::item.url }}"></a>
+                                </div>
+                                <div class="col-4">
+                                    <a target="_blank"
+                                        class="icon-dropdown icon-twitter pr-5"
+                                        ng-href="https://twitter.com/intent/tweet?url={{ ::item.url }}"></a>
+                                </div>
```

Comment:
Can the social media icons be made conditional on the existence of the relevant share app?

Reply:
Since I just found it here a list of the currently available socialsharing apps: [nextcloud/socialsharing](https://github.com/nextcloud/socialsharing).

Not sure of the implementation there is useful for you here. (e.g. https://github.com/nextcloud/socialsharing/blob/d03ba4f3b9db178051be661b2607ce9c8930f396/socialsharing_email/js/socialsharingemail.js#L24-L32)

Reply:
Thanks for the link and your help!

We've already found and studied this option. Our idea is to use the OCA.Sharing.ExternalLinkActions (like it's done in the file sharing app), and loop over the actions.
However, we do not have access to OCA.Sharing. We've added the OCA global in the `.jshintrc` file, which gives us access to OCA, however it doesn't contain `Sharing`. We're still looking into it to figure out why, and we've also posted a [question](https://help.nextcloud.com/t/how-to-access-oca-sharing/110799) on the forum just in case we take too long to find the solution.